src/connector/src/source/error.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::backtrace::Backtrace;
+use std::num::ParseIntError;
+use std::sync::Arc;
+
+use rdkafka::error::KafkaError;
+use risingwave_common::array::ArrayError;
+use risingwave_pb::ProstFieldNotFound;
+
+pub type SourceResult<T> = std::result::Result<T, SourceError>;
+
+#[derive(thiserror::Error, Debug)]
+enum SourceErrorInner {
+    #[error("Source error: {0}")]
+    SourceError(String),
```

Comment:
What's the semantic of `SourceError::SourceError`? 🥵

Reply:
I think it's just `InternalError`? In other components, we use `InternalError` to represent error originated from this module.